crime-evidence/src/test/java/uk/gov/justice/laa/crime/evidence/controller/IncomeEvidenceControllerTest.java

```diff
@@ -0,0 +1,103 @@
+package uk.gov.justice.laa.crime.evidence.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpMethod;
+import org.springframework.test.web.servlet.MockMvc;
+import uk.gov.justice.laa.crime.common.model.evidence.ApiCreateIncomeEvidenceRequest;
+import uk.gov.justice.laa.crime.common.model.evidence.ApiUpdateIncomeEvidenceRequest;
+import uk.gov.justice.laa.crime.commons.tracing.TraceIdHandler;
+import uk.gov.justice.laa.crime.evidence.data.builder.TestModelDataBuilder;
+import uk.gov.justice.laa.crime.util.RequestBuilderUtils;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(IncomeEvidenceController.class)
+@AutoConfigureMockMvc(addFilters = false)
+class IncomeEvidenceControllerTest {
+
+    private static final String ENDPOINT_URL = "/api/internal/v1/evidence";
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private TraceIdHandler traceIdHandler;
+
+    @Test
+    void createEvidence_ServerError_RequestBodyIsMissing() throws Exception {
+        mvc.perform(RequestBuilderUtils.buildRequestGivenContent(HttpMethod.POST, "", ENDPOINT_URL))
+                .andExpect(status().is4xxClientError());
+    }
+
+    @Test
+    void createEvidence_BadRequest_RequestEmptyBody() throws Exception {
+        mvc.perform(RequestBuilderUtils.buildRequestGivenContent(HttpMethod.POST, "{}", ENDPOINT_URL))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void createEvidence_Unauthorized_NoAccessToken() throws Exception {
+        mvc.perform(RequestBuilderUtils.buildRequestGivenContent(HttpMethod.POST, "{}", ENDPOINT_URL, false))
+                .andExpect(status().is4xxClientError());
```

Comment:
Maybe make this a more specific status that you're expecting. If for any reason the auth isn't doing it's thing, you will still get back a 4xx response as the request then is pretty much identical to the previous test that has no request body, but the test would still pass.

Reply:
Great point. It turns out when changing it to `isUnauthorised()` the tests failed because the `springSecurityFilterChain` is disabled in the Controller tests. Therefore, I've moved these scenarios over to the integration test classes and took the opportunity to do a little refactoring and created an `IntegrationTestBase` that contains shared configuration and stubs, which should help to cut down on duplication.

Reply:
Nice one, sounds great!